lib/src/source/git.dart

```diff
@@ -253,9 +302,20 @@ class GitSource extends CachedSource {
   /// by [id] on the effective ref of [id].
   ///
   /// This assumes that the canonical clone already exists.
-  Future<String> _getRev(PackageId id) {
-    return git.run(["rev-list", "--max-count=1", _getEffectiveRef(id)],
-        workingDir: _repoCachePath(id)).then((result) => result.first);
+  Future<String> _getRev(PackageId id) async {
+    var ref = _getEffectiveRef(id);
+    return git
+        .run(["rev-list", "--max-count=1", ref], workingDir: _repoCachePath(id))
+        .then((result) => result.first)
```

Comment:
In general, yes. It's easier to read and our debugger actually has special support for async/await – that we don't get if you use the Future API directly.

@nex3 has found a few cases where it's better to use the API directly, but she'd have to comment.